### PR TITLE
Move and revise overview for Security section

### LINKED ISSUE #36591: Update or remove https://kubernetes.io/docs/concepts/security/overview/
**This is a Refactor request**

**Problem:**
The overview page at https://kubernetes.io/docs/concepts/security/overview/ is a bit out of date:
- no reference to https://kubernetes.io/docs/concepts/security/security-checklist/
- I'm not convinced that the “Four Cs of cloud native security” model is useful for readers to learn
- **Does not align with other CNCF documents, such as https://github.com/cncf/tag-security/tree/main/security-whitepaper**

**Proposed Solution:**
Either:
- remove https://kubernetes.io/docs/concepts/security/overview/ outright
- migrate content from https://kubernetes.io/docs/concepts/security/overview/ into several destinations
  - https://kubernetes.io/docs/concepts/security/ _(keep this brief but on-point)_
  - https://kubernetes.io/docs/concepts/security/security-checklist/
  - https://kubernetes.io/docs/concepts/security/controlling-access/
  
  &hellip;and then redirect https://kubernetes.io/docs/concepts/security/overview/ to https://kubernetes.io/docs/concepts/security/

**Page to Update:**
https://kubernetes.io/docs/concepts/security/overview/

**Additional Information:**
/sig docs
/sig security
/language en

### PR DESCRIPTION
**Pull request by sftim**
A fix for https://github.com/kubernetes/website/issues/36591 (fixes #36591)

- Change https://kubernetes.io/docs/concepts/security/ to be an overview of security ([preview](https://deploy-preview-43176--kubernetes-io-main-staging.netlify.app/docs/concepts/security/))
- Add another page about cloud native security and Kubernetes, based on the CNCF security white paper ([preview](https://deploy-preview-43176--kubernetes-io-main-staging.netlify.app/docs/concepts/security/cloud-native-security/))
- Remove https://kubernetes.io/docs/concepts/security/overview/ (with a redirect)

We should make this change because https://kubernetes.io/docs/concepts/security/overview/ is not really giving readers a good overview of how to secure a Kubernetes cluster. That page was much better than no advice; this PR aims to improve it further by explicitly aligning with the CNCF [security white paper](https://www.cncf.io/reports/cloud-native-security-whitepaper/).

/language en
/sig security

### CODE REVIEW

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [inline content/en/docs/concepts/security/_index.md:216]: This relies on `simple_list: true` in the front matter. See the [preview](https://deploy-preview-43176--kubernetes-io-main-staging.netlify.app/docs/concepts/security/).

**Review by Gauravpadam — COMMENTED** [review]
Thanks,

A partial review from my end

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md:34]: I don't think so. Reviewing docs isn't really relevant to securing a cluster.

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md:115]: No, because the glossary tooltip does that automatically.

File: content/en/docs/concepts/security/_index.md
Comment (aj11anuj) [inline content/en/docs/concepts/security/_index.md:149]:
```suggestion
1. Protect data durability using backups. Verify that you can restore these, whenever you need to.
```

File: content/en/docs/concepts/security/_index.md
Comment (aj11anuj) [inline content/en/docs/concepts/security/_index.md:151]:
```suggestion
   upon it.
```

File: content/en/docs/concepts/security/_index.md
Comment (aj11anuj) [inline content/en/docs/concepts/security/_index.md:130]:
```suggestion
   mechanisms such as [LimitRange](/docs/concepts/policy/limit-range/)
```

File: content/en/docs/concepts/security/_index.md
Comment (aj11anuj) [inline content/en/docs/concepts/security/_index.md:135]:
```suggestion
   the Pods with different trust contexts will run on separate sets of node.
```

File: content/en/docs/concepts/security/_index.md
Comment (aj11anuj) [inline content/en/docs/concepts/security/_index.md:110]:
```suggestion
trade-offs in finding an appropriate balance.
```

File: content/en/docs/concepts/security/_index.md
Comment (aj11anuj) [inline content/en/docs/concepts/security/_index.md:99]:
```suggestion
TLS (including the traffic between nodes and control plane), and protect the
```

File: content/en/docs/concepts/security/_index.md
Comment (aj11anuj) [inline content/en/docs/concepts/security/_index.md:102]:
```suggestion
pay special attention for restricting misuse there.
```

File: content/en/docs/concepts/security/_index.md
Comment (aj11anuj) [inline content/en/docs/concepts/security/_index.md:86]:
```suggestion
The Kubernetes API is what makes your cluster work. Protecting this API is a key
for providing effective cluster security.
```

File: content/en/docs/concepts/security/_index.md
Comment (aj11anuj) [inline content/en/docs/concepts/security/_index.md:79]:
```suggestion
The Runtime phase comprises of three critical areas: [compute](#protection-runtime-compute),
```

File: content/en/docs/concepts/security/_index.md
Comment (aj11anuj) [inline content/en/docs/concepts/security/_index.md:67]:
```suggestion
and where it can be deployed.
```

File: content/en/docs/concepts/security/_index.md
Comment (aj11anuj) [inline content/en/docs/concepts/security/_index.md:14]:
```suggestion
{{< glossary_tooltip text="CNCF" term_id="cncf" >}} about good practices for
```

**Review by aj11anuj — CHANGES_REQUESTED** [review]
Some minor suggestions from my side

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md:151]: I'm afraid I won't accept this suggestion as it doesn't look like typical English grammar.

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md:130]: I'm afraid I won't accept this suggestion as it would look odd to have ResourceQuotas in plural and LimitRange in singular.

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md:135]: I'm afraid I won't accept this suggestion as it doesn't look like idiomatic English grammar.

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md:110]: I'm afraid I prefer my text.

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md:102]: I'm afraid I prefer my text. Was there a reason you proposed an edit there?

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md:99]: I wouldn't feel happy putting my name to that sentence.

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md:86]: I'm afraid I won't accept this suggestion as it doesn't look like idiomatic English grammar.

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md:79]: I'm afraid I won't accept this suggestion as it doesn't look like idiomatic English grammar.

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md:67]: I (still) think “where is can be deployed to” is how people who speak English would say it.

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md:14]: I could make this change, but: I'm also happy with the original wording. It doesn't feel like this is a blocker to merging the PR.

**Comment by sftim** [comment]
Thanks for the review @aj11anuj. However, please watch out for making suggestions that change text from good use of English to invalid or unidiomatic English.

**Comment by sftim** [comment]
Part of https://github.com/kubernetes/website/issues/25119

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md]: How's this now?

File: content/en/docs/concepts/security/_index.md
Comment (divya-mohan0209) [reply content/en/docs/concepts/security/_index.md:71]: +1 to what Tim said.

File: content/en/docs/concepts/security/_index.md
Comment (divya-mohan0209) [reply content/en/docs/concepts/security/_index.md:38]: +1 to what Tim said.

**Comment by sftim** [comment]
@divya-mohan0209 as you've taken a look already, would you be willing to look again?

As is often our favored approach, I'm looking to find the smallest set of changes that could let this merge. I'm also very happy to hear any non-binding feedback that I might then adopt or might save for a separate PR.

File: content/en/docs/concepts/security/_index.md
Comment (kbhawkey) [inline content/en/docs/concepts/security/_index.md:37]: minimizes

File: content/en/docs/concepts/security/_index.md
Comment (kbhawkey) [inline content/en/docs/concepts/security/_index.md:89]: nit: This sentence is a bit general.

**Comment by kbhawkey** [comment]
I reviewed the preview (https://deploy-preview-43176--kubernetes-io-main-staging.netlify.app/docs/concepts/security/)
It's not clear that the link to the Security certification program and the other links to the CNCF white papers should be part of this overview page.

File: content/en/docs/concepts/security/_index.md
Comment (kbhawkey) [inline content/en/docs/concepts/security/_index.md:178]: Can you fix the spelling error in the heading?

File: content/en/docs/concepts/security/_index.md
Comment (kbhawkey) [inline content/en/docs/concepts/security/_index.md:231]: I suggest to remove these links and the link to the certification course (seems like advertising).

Comment (sftim) [reply content/en/docs/concepts/security/_index.md:231]: I think we'd quite like to advertise the course - it's how we (our foundation, the CNCF) make some of the revenue that pays for running the website.

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md]: Sure!

**Comment by sftim** [comment]
> I reviewed the preview (https://deploy-preview-43176--kubernetes-io-main-staging.netlify.app/docs/concepts/security/) It's not clear that the link to the Security certification program and the other links to the CNCF white papers should be part of this overview page.

I've separated out the CNCF white paper links. I still support linking to the CNCF program; as Kubernetes is a CNCF project, I'm confident it's fine.

File: content/en/docs/concepts/security/_index.md
Comment (nate-double-u) [reply content/en/docs/concepts/security/_index.md]: I agree with @sftim here, the original wording is more grammatically correct.

**Comment by PushkarJ** [comment]
@sftim I read the updates at a high level and they read (technically) right to me. There maybe minor things we need to change which I missed but that could be accomplished in follow up PRs.

Few things that I recommend we should consider updating before we merge: 
- We should link existing seccomp profile k8s documentation. I didn't see it linked in either of the two pages, but I could have used it. 
- Highly request linking to cloud native security white paper using this link instead of current link: https://github.com/cncf/tag-security/tree/main/security-whitepaper as it has the pdf too but more importantly most up to date markdown where minor corrections post publication were made. 
- Optionally, if you see the above link you will find the paper translated into multiple languages. Since we maintain k8s docs for many languages too we could consider making a special note of the translated white paper if one exists for that particular language. This will be a wonderful user experience in my opinion. 

Notes: 
- I reviewed this on GitHub mobile app so don't know how it reads on a laptop screen 
- I was the lead author for version 2 of the white paper, so I appreciate this massive lift from SIG Docs to make the paper relevant for our community!

**Comment by sftim** [comment]
> We should link existing seccomp profile k8s documentation. I didn't see it linked in either of the two pages, but I could have used it.

I recommend covering that in a follow-up PR. The content I'm replacing didn't link there either.

---

> Highly request linking to cloud native security white paper using this link instead of current link: https://github.com/cncf/tag-security/tree/main/security-whitepaper as it has the pdf too but more importantly most up to date markdown where minor corrections post publication were made.

OK, will update.

---

> Optionally, if you see the above link you will find the paper translated into multiple languages. Since we maintain k8s docs for many languages too we could consider making a special note of the translated white paper if one exists for that particular language. This will be a wonderful user experience in my opinion.

I'll add a note to localizers.

**Comment by PushkarJ** [comment]
Looks good to me now. 

Hopefully one more person from sig security can review too

File: content/en/docs/concepts/security/_index.md
Comment (tabbysable) [inline content/en/docs/concepts/security/_index.md:28]:
```suggestion
A key security mechanism for any Kubernetes cluster is to
```
Nit

File: content/en/docs/concepts/security/_index.md
Comment (tabbysable) [inline content/en/docs/concepts/security/_index.md:31]: Nit: Recommend calling out TLS first in this paragraph, since it is essential to virtually any effort to secure a kubernetes cluster. Then move to "You can also configure encryption at rest..." because it's a good idea but less critical for many common use cases.

File: content/en/docs/concepts/security/cloud-native-security.md
Comment (tabbysable) [inline content/en/docs/concepts/security/cloud-native-security.md:117]: Just wanna say that I LOVE this sentence especially.

File: content/en/docs/concepts/security/cloud-native-security.md
Comment (tabbysable) [inline content/en/docs/concepts/security/cloud-native-security.md:128]:
```suggestion
   for applications, to help ensure they run with only the necessary privileges.
```

Comment (tabbysable) [reply content/en/docs/concepts/security/cloud-native-security.md:128]: This is a nit, but I think the change in tone here is important to encourage the reader in the right directions.

**Comment by tabbysable** [comment]
I left several nits, none of which should be interpreted as merge-blocking.

This is beautiful! Thank you for the huge amount of thoughtful work that has gone into it.

/lgtm

Please ping me to re-LGTM if you choose to address any of the nits and the LGTM thus falls off.

File: content/en/docs/concepts/security/_index.md
Comment (AnshumanTripathi) [inline content/en/docs/concepts/security/_index.md:39]: Should we add that secrets are only base64 encoded and are stored unencrypted in the API server? (A premonition to the caution block under the secrets documentation)

File: content/en/docs/concepts/security/cloud-native-security.md
Comment (AnshumanTripathi) [inline content/en/docs/concepts/security/cloud-native-security.md:77]: Nit: We are using artifacts (american spelling) everywhere else
```suggestion
cryptographic identity of container image artifacts.
```

File: content/en/docs/concepts/security/cloud-native-security.md
Comment (AnshumanTripathi) [inline content/en/docs/concepts/security/cloud-native-security.md:79]:
```suggestion
When you deploy Kubernetes, you also set the foundation for your
```
`When you deploy Kubernetes` seems confusing. Do mean `Deploy to Kubernetes` or `Deploy a Kubernetes cluster`?

**Comment by AnshumanTripathi** [comment]
Missed to comment lgtm

/lgtm

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md:41]: Maybe you'd like to propose that for a follow-up PR?

**Comment by sftim** [comment]
@tabbysable, I think this is ready for another LGTM if you're happy with it.

File: content/en/docs/concepts/security/_index.md
Comment (sftim) [reply content/en/docs/concepts/security/_index.md]: (reworded)

File: content/en/docs/concepts/security/cloud-native-security.md
Comment (sftim) [reply content/en/docs/concepts/security/cloud-native-security.md:79]: I mean deploy a Kubernetes cluster; is that not as obvious as I'd've hoped?

Comment (AnshumanTripathi) [reply content/en/docs/concepts/security/cloud-native-security.md:79]: I guess it makes sense. 👍

**Comment by PushkarJ** [comment]
/lgtm

( @tabbysable hoping to save you some keystrokes )

**Comment by reylejano** [comment]
This PR improves the security content in the docs
Thank you
/approve

**Pull request merged**
